src/buffer.rs

```diff
@@ -157,25 +231,25 @@ impl AudioBuffer {
     /// Resample to the desired sample rate.
     ///
     /// This changes the sample_length of the buffer.
-    ///
-    /// ```
-    /// use web_audio_api::SampleRate;
-    /// use web_audio_api::buffer::{ChannelData, AudioBuffer};
-    ///
-    /// let channel = ChannelData::from(vec![1., 2., 3., 4., 5.]);
-    /// let mut buffer = AudioBuffer::from_channels(vec![channel], SampleRate(48_000));
-    ///
-    /// // upmix from 48k to 96k Hertz sample rate
-    /// buffer.resample(SampleRate(96_000));
-    ///
-    /// assert_eq!(
-    ///     buffer.channel_data(0),
-    ///     &ChannelData::from(vec![1., 1., 2., 2., 3., 3., 4., 4., 5., 5.,])
-    /// );
-    ///
-    /// assert_eq!(buffer.sample_rate().0, 96_000);
-    /// ```
-    pub fn resample(&mut self, sample_rate: SampleRate) {
+    //
+    // ```
+    // use web_audio_api::SampleRate;
+    // use web_audio_api::buffer::{ChannelData, AudioBuffer};
+    //
+    // let channel = ChannelData::from(vec![1., 2., 3., 4., 5.]);
+    // let mut buffer = AudioBuffer::from_channels(vec![channel], SampleRate(48_000));
+    //
+    // // upmix from 48k to 96k Hertz sample rate
+    // buffer.resample(SampleRate(96_000));
+    //
+    // assert_eq!(
+    //     buffer.get_channel_data(0)[..],
+    //     vec![1., 1., 2., 2., 3., 3., 4., 4., 5., 5.][..]
+    // );
+    //
+    // assert_eq!(buffer.sample_rate().0, 96_000);
+    // ```
+    pub(crate) fn resample(&mut self, sample_rate: SampleRate) {
```

Comment:
We should definitely replace this version later with the library we are already using

Reply:
you mean the `rubato` lib here?

Reply:
yes!